packages/data-stores/src/user/types.ts

```diff
@@ -1,44 +1,8 @@
 import * as selectors from './selectors';
 import type { ActionCreators } from './actions';
 import type { SelectFromMap } from '../mapped-types';
-export interface CurrentUser {
-	ID: number;
-	display_name: string;
-	username: string;
-	email: string;
-	/**
-	 * The user's locale slug, e.g. `es`.
-	 */
-	language: string;
-
-	/**
-	 * The bootstraped user's locale slug, e.g. `es`.
-	 * @deprecated Use `language` instead.
-	 */
-	localeSlug?: string;
-
-	/**
-	 * The user's locale variant, e.g. `es-mx`.
-	 * If there is no variant, `""` empty string is returned.
-	 */
-	locale_variant: string;
-
-	/**
-	 * The bootstrapped user's locale variant, e.g. `es-mx`.
-	 * @deprecated Use `locale_variant` instead.
-	 */
-	localeVariant?: string;
-
-	/**
-	 * The user's existing sites count.
-	 */
-	site_count: number;
-
-	/**
-	 * The subkey for Subscription Management
-	 */
-	subscriptionManagementSubkey?: string;
-}
+import type { User } from '@automattic/api-core';
 
+export type CurrentUser = Partial< User > & Pick< User, 'ID' | 'display_name' | 'email' >;
```

Comment:
There's an issue here where v2 treats many of the fields as required, but the type of `currentUser` here still says that most of the fields are optional.

Maybe we rationalise these types by checking that the fields really are reliably bootstrapped, and then marking them as non-optional?